src/main/java/servlets/VendorServlet.java

```diff
@@ -33,12 +45,73 @@
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     JsonConverter converter = new JsonConverter();
     String configMap = converter.getConfigMap();
-    response.setContentType("text/html;");
+    response.setContentType(CONTENT_TYPE);
     response.getWriter().println(configMap);
   }
 
-  @Override
-  public void doDelete(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // DELETE method --> deletes vendor configuration directory
+  /** Delete the desired Vendor from the fileset and spreadsheets.*/
+  public void doDelete(HttpServletRequest request, HttpServletResponse response) throws IOException {  
+    String vendorID = request.getParameter(VENDOR_ID_PARAM);
+    String responseString = "success"; 
+
+    // Delete the file from the filesystem and update the sheets.
+    try {
+      deleteVendorFile(vendorID);
+      updateSheets(request);    
+    } catch (IOException e) {
+      responseString = "Failed to Delete File.";
+    } catch (GeneralSecurityException e) {
+      responseString = "Failed to Delete File.";
+    } 
+
+    // Convert the String message into a JSON String.
+    String jsonMessage = messageAsJson(responseString);
+
+    response.setContentType(CONTENT_TYPE);
+    response.getWriter().println(jsonMessage);
+  }
+
+  private void deleteVendorFile(String vendorID) throws FileNotFoundException {
+    JsonConverter converter = new JsonConverter();
+    ArrayList<String> vendorIDs = converter.getVendorIDs();
+
+    // Validate that the vendorID exists.
+    if(!vendorIDs.contains(vendorID)) {
+        throw new FileNotFoundException();
+    }
+    converter.deleteFile(vendorID);
+  }
+
+  /** Rebuild the sheets without the deleted Vendor's data. */
+  private void updateSheets(HttpServletRequest request) 
+      throws IOException, GeneralSecurityException {
+    JsonConverter converter = new JsonConverter();
+    ArrayList<String> vendorIDs = converter.getVendorIDs();
+    ArrayList<Vendor> vendors = getVendors(vendorIDs, converter); 
+
+    // Only retrieve the session if one exists.
+    HttpSession session = request.getSession(false);
+    String accessToken = session.getAttribute("accessToken").toString();
```

Comment:
Can we use a constant for "accessToken"?

Reply:
sure